mlir/lib/Quantum/Transforms/ChainedSelfInversePatterns.cpp

```diff
@@ -56,9 +56,15 @@ struct ChainedNamedHermitianOpRewritePattern : public mlir::OpRewritePattern<Cus
 
         // Replace uses
         ValueRange InQubits = op.getInQubits();
-        auto ParentOp = dyn_cast_or_null<CustomOp>(InQubits[0].getDefiningOp());
-        ValueRange simplifiedVal = ParentOp.getInQubits();
-        rewriter.replaceOp(op, simplifiedVal);
+        auto parentOp = dyn_cast_or_null<CustomOp>(InQubits[0].getDefiningOp());
+        ValueRange originalNonCtrlQubits = parentOp.getNonCtrlQubitOperands();
+        ValueRange originalCtrlQubits = parentOp.getCtrlQubitOperands();
```

Comment:
Are method calls safe on a possible null?

Reply:
Good point, the verification analysis above already guarantees at this point the parent will be of the same type (`CustomOp`).

Reply:
done